sentinel/lib/lineage.js

```diff
@@ -47,31 +52,55 @@ const minifyContact = contact => {
   return result;
 };
 
-const fetchHydratedDoc = id => {
-  return new Promise((resolve, reject) => {
+const patientLineageByShortcode = shortcode =>
+  new Promise((resolve, reject) => utils.getPatientContactUuid(db, shortcode, (err, uuid) => {
+    if (err) {
+      reject(err);
+    } else {
+      lineageById(uuid).then(resolve);
```

Comment:
I think you still have a problem here if `lineageById` fails the error will get lost. I think you either need to catch and reject or find some other way to pass the error out.